src/wovnio/wovnphp/CookieLang.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+
+namespace Wovnio\Wovnphp;
+
+
+class CookieLang
+{
+    const COOKIE_LANG_NAME = 'wovn_selected_lang';
+
+    private $header;
+    private $store;
+
+    public function __construct($header, $store)
+    {
+        $this->header = $header;
+        $this->store = $store;
+        $this->setCookieIfNeeded();
+    }
+
+    public function setLangCookie($langCode)
+    {
+        if (!array_key_exists($langCode, Lang::$index)) {
+            return;
+        }
+
+        setcookie(CookieLang::COOKIE_LANG_NAME, $langCode, time()+60*60*24*30, '/', $this->header->host, true);
+    }
+
+    public static function getCookieLang()
+    {
+        if (array_key_exists(CookieLang::COOKIE_LANG_NAME, $_COOKIE)) {
+            return $_COOKIE[CookieLang::COOKIE_LANG_NAME];
+        }
+        return null;
+    }
+
+    public function shouldRedirect() {
+        return self::getCookieLang() && ($this->header->lang() != self::getCookieLang());
+    }
+
+    public function computeRedirectUrl()
```

Comment:
Feel like this should not be the responsibility of CookieLang.

Reply:
in this case Headers.responseOut probably _is_ the correct place for dealing with redirects.

Reply:
Indeed it shouldn't. The logic were taken directly out of HtmlConverter. I tried to make it public from HtmlConverter but its requiring way too many constructor arguments to be practical in CookieLang. Perhaps it's better if I move it to Url?

Reply:
This is mostly delegating to headers methods here, which is why I think headers is the best place for it. Url is supposed to only be about manipulating lang codes on URLs.